2018/2018_File_Generation_Features.md

```diff
@@ -0,0 +1,67 @@
+
+Each edit logic method in `lar_constraints`, `test_data`, and `rules_engine` is prefaced with 's', 'v', or 'q' to indicate a syntax, validity or quality edit respectively.
+
+### Validating Quality Edit Files and Creating Edit Reports
+The `test_data` logic for creating quality edit test files may cause rows to fail not only quality edits, but also syntax or validity edits in the creation process. One method in the FileGenerator class, `validate_quality_edit_file(quality_filename)` takes a quality edit test file from the quality filepaths directory in `test_filepaths.yaml`, drops rows that have syntax or validity edits, and duplicates the remaining clean rows to the length of the original file. The file is then saved in a new directory.
+
+FilesGenerator also contains a method, `edit_report` that produces an edit report for a test file which is stored locally as a csv file. The edit report csv is stored in a directory and with a filename specified in `test_filepaths.yaml` 
+
+## The Large Test Files Script
+The Large Test Files script in [`large_test_files_script.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/large_test_files_script.py) creates a test file based on existing test file data to create a new file with a larger number of rows. The script would only need to be run, and the attributes of the new file to be created can be changed with [`large_file_script_config.yaml`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/large_file_script_config.yaml) in `python/configurations`. Files can be created with any number of rows, and can be made for any test institution so long as the Institution Name, LEI and Tax ID are provided in the configuration file. The configuration file stores filepaths for source data, as well as the filepaths for output. 
+
+In addition, the script contains functionality for changing values by column and row using a yaml configuration. An example of this yaml configuration is located in `python/configurations/` as 
+[`row_by_row_modification.yaml.`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/configurations/row_by_row_modification.yaml) The script can contain a path to a yaml configuration to initiate this functionality. 
+
+
+## Additional Utilities
+In addition to test file creation features, the repository contains utility functions that work with existing test file data. These functions assist in creating test files with further customization, such as files with a large number of rows or files as part of a new institution. The functions are located in [`python/utils.py`](https://github.com/cfpb/hmda-test-files/blob/master/2018/python/utils.py). 
+
+These functions include:
+- `write_file(path, ts_input, lar_input, name="test_file.txt")` 
```

Comment:
Add a clarifying line that a submission file a single TS row and one to N LAR rows.